.github/workflows/continous-integration.yml

```diff
@@ -155,13 +157,52 @@ jobs:
         COVERALLS_SERVICE_NAME: github
       run: poetry run coveralls
 
+  docker:
+    name: Build Docker
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        image:
+          - {"file": "Dockerfile", "tag_ext": ""}
+          - {"file": "docker/Dockerfile_full", "tag_ext": "-full"}
+          - {"file": "docker/Dockerfile_pretrained_embeddings_mitie_en", "tag_ext": "-mitie-en"}
+          - {"file": "docker/Dockerfile_pretrained_embeddings_spacy_de", "tag_ext": "-spacy-de"}
+          - {"file": "docker/Dockerfile_pretrained_embeddings_spacy_en", "tag_ext": "-spacy-en"}
+
+    env:
+      DOCKERHUB_USERNAME: tmbo
+
+    steps:
+    - name: Checkout git repository 🕝
+      uses: actions/checkout@v2
+
+    - name: Login to DockerHub Registry 🔢
+      run: echo ${{ secrets.DOCKERHUB_PASSWORD }} | docker login -u ${{ env.DOCKERHUB_USERNAME }} --password-stdin
+
+    - name: Pull latest${{ matrix.image.tag_ext }} Docker image for caching
```

Comment:
do we need to pull this explicitly or is docker clever enough to do so? Should probably also add the pulling part to Rasa X then.